cmd/eksctl-anywhere/cmd/listpackages.go

```diff
@@ -0,0 +1,61 @@
+package cmd
+
+import (
+	"context"
+	"log"
+
+	"github.com/spf13/cobra"
+
+	"github.com/aws/eks-anywhere/pkg/curatedpackages"
+	"github.com/aws/eks-anywhere/pkg/kubeconfig"
+)
+
+type listPackagesOption struct {
+	kubeVersion string
+	source      string
+}
+
+var lpo = &listPackagesOption{}
+
+func init() {
+	listCmd.AddCommand(listPackagesCommand)
+	listPackagesCommand.Flags().StringVar(&lpo.source, "source", "", "Discovery Location. Options (cluster, registry)")
```

Comment:
You could use a custom flag here to handle validation for you in a consistent way.

https://cs.opensource.google/go/go/+/go1.18:src/flag/flag.go;l=893

Reply:
@ewollesen, I am not sure I understand this comment. I am currently using StringVar to handle the validation. Are you suggesting to use Var instead of StringVar for the validation?